pkg/agent/controller/agent.go

```diff
@@ -381,6 +385,20 @@ func (a *Controller) serviceToRemoteServiceImport(obj runtime.Object, numRequeue
 	serviceImport := a.newServiceImport(svcExport.Name, svcExport.Namespace)
 
 	// Update the status and requeue
+	if op == syncer.Update {
+		err := a.serviceImportSyncer.GetLocalFederator().Delete(serviceImport)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, true
+		}
+		// some other error. Log and requeue
+		klog.Errorf("Error delete ServiceImport for Service (%s/%s): %v", svc.Namespace, svc.Name, err)
+
+		a.updateExportedServiceStatus(svcExport.Name, svcExport.Namespace, corev1.ConditionFalse, serviceUpdate,
+			"Service update")
+
+		return nil, false
```

Comment:
What specific modification of an exported service would warrant this action? It seems potentially disruptive to delete the `ServiceImport` - why not update the existing one?

Reply:
The service may change to an unsupported type service. service Import seem don't have a not avalible status

Reply:
Then let's handle that specific case rather than blindly deleting the `ServiceImport`.

Reply:
Good idea, i will try to solve it in the next few days

Reply:
update `ServiceExport` status in Service deletion, let `ServiceExport -> ServiceImport` decide how to deal with it.